### PR TITLE
feat(config): use json for project configs

### PR DESCRIPTION
**Pull request by pfongkye**
- Use a JSON file to store project configs so that it's easier for each Homer user to add the config file in the correct directory before deploying Homer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -306,6 +306,25 @@ displayed.
 
 You should now be able to use Homer commands on Slack 🎉
 
+### 7. Add a JSON file for the release configurations
```
Comment (greg0ire) [inline README.md:309]: Let's keep the title free from implementation details.

```suggestion
### 7. Configure releases
```

File: README.md [hunk 2]
```diff
@@ -306,6 +306,25 @@ displayed.
...
+
+Add/update a JSON file (it must be called `projects.json`) in `config/homer` folder to enable release management for your projects. The file should be like below:
```
Comment (greg0ire) [inline README.md:311]:
```suggestion
Releases management can be configured in `config/homer/projects.json`.
Here is a sample configuration with one project:
```

File: __tests__/release/utils/configBuilder.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      },
+      {
+        notificationChannelIds: ['C678'],
+        projectId: 890,
+        releaseChannelId: 'C456',
+        releaseManager: defaultReleaseManager,
+      },
+    ]);
+  });
+  it('should return empty array if configs is not an array', () => {
```
Comment (greg0ire) [inline __tests__/release/utils/configBuilder.test.ts:44]: Should it throw instead?

File: __tests__/release/utils/configBuilder.test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    expect(
+      buildProjectReleaseConfigs(
+        {} as any,
+        { defaultReleaseManager },
+        { federationReleaseTagManager }
+      )
+    ).toEqual([]);
+  });
+  it('should return empty array if no valid configurations', () => {
```
Comment (greg0ire) [inline __tests__/release/utils/configBuilder.test.ts:53]: Same here?

File: src/release/utils/configBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import type { JSONSchemaType } from 'ajv';
+import Ajv from 'ajv';
+import type {
+  ProjectReleaseConfig,
+  ProjectReleaseConfigJSON,
+} from '../typings/ProjectReleaseConfig';
+import type { ReleaseManager } from '../typings/ReleaseManager';
+import type { ReleaseTagManager } from '../typings/ReleaseTagManager';
+
+// only one instance should be used across all project
```
Comment (greg0ire) [inline src/release/utils/configBuilder.ts:10]: Do you mean homer? => "all the project"
Or do you mean all the projects in the config file? => "all projects"

File: __tests__/release/utils/configBuilder.test.ts (hunk 3 above)
Comment (pfongkye) [reply __tests__/release/utils/configBuilder.test.ts:44]: Yes I think it is better to let the user know that something is wrong with the configuration.

**Comment by fabienfleureau** [comment]
Hello, with this evolution would it be possible to have the json injected after the build of the project?
For example, homer is built and packaged in a docker image, I can mount a volume with the config file at the right place to have it loaded at startup.

**Review by greg0ire — APPROVED** [review]

**Comment by pfongkye** [comment]
> Hello, with this evolution would it be possible to have the json injected after the build of the project? For example, homer is built and packaged in a docker image, I can mount a volume with the config file at the right place to have it loaded at startup.

Hello @fabienfleureau,
For now we are building a new image with the config json copied to the correct destination but I think it is a good idea also to be able to mount a volume so that we only need to update the config file when necessary. I haven't tried it but could be a good evolution also. Let us know if you want to try this approach.
Thanks.

**Pull request merged**